### PR TITLE
Add best per instance scaling to scaling results

### PR DESCRIPTION
**Pull request by bodono**
Update the scaling plots to show best on a per-instance basis.

### CODE REVIEW

File: scripts/analyze_csv_data.py [hunk 1]
```diff
@@ -681,9 +684,23 @@ def gen_all_improvement_plots(outputs, prefix):
 # bisco scaling vs no scaling (NO JOIN DEFAULT)
 df = pd.read_csv(os.path.join(CSV_DIR, 'miplib_scaling_100k.csv'))
 df = fill_in_missing_problems(df, miplib_instances)
+
+# Pull out best performing scaling for each instance / tolerance:
```
Comment (dapplegate) [inline scripts/analyze_csv_data.py:688]: Pre-existing: I believe the comment at line 605 should be "Pull out best step-size rule for each instance...".

**Review by dapplegate — APPROVED** [review]

File: scripts/analyze_csv_data.py (hunk 1 above)
Comment (bodono) [reply scripts/analyze_csv_data.py:688]: Fixed, thanks.

**Comment by bodono** [comment]
Btw, if we merge this then the figures produced by the script will no longer match the paper we submitted. We should decide if we want to change the paper figs, or if we don't want this change at all in the main script, or we do want it, but we want the original paper figs as well (would need to update the script for that).

**Comment by dapplegate** [comment]
> Btw, if we merge this then the figures produced by the script will no longer match the paper we submitted. We should decide if we want to change the paper figs, or if we don't want this change at all in the main script, or we do want it, but we want the original paper figs as well (would need to update the script for that).

Oops, I missed that.  I think that at least for now, we should make sure we still generate the plots in the paper, so please update the script to keep the old plots as well as generate the new one.

**Review by dapplegate — CHANGES_REQUESTED** [review]
Withdrew approval, after comment pointing out that after this change, the script would no longer generate the plots from the draft paper.

**Comment by bodono** [comment]
Updated to generate original figs / tables under original names and new figs / tables under new names.

**Review by dapplegate — APPROVED** [review]

**Pull request merged**